scss/components/dialog.scss

```diff
@@ -0,0 +1,121 @@
+@import "./../settings";
+@import "./../mixins";
+@import "./../functions";
+@import "./../icons/mixins";
+
+/*!
+.fd-dialog
+    .fd-dialog__header
+        .fd-form__title
+    .fd-form__body
+    .fd-form__footer-items
+      .fd-dialog__button-primary
+      .fd-dialog__button-secondary
+*/
+
+
+$block: #{$fd-namespace}-dialog;
+  $fd-dialog-width--small: 288px !default;
+  $fd-dialog-width--medium: 460px !default;
+  $fd-dialog-width--phone: 320px !default;
+  $fd-dialog-header-height: 48px !default;
+  $fd-dialog-footer-height: 48px !default;
+
+  $fd-dialog-border-color: fd-color("neutral", 2) !default;
+  $fd-dialog-padding-x: var(--fd-width-gutter) !default;
+  $fd-dialog-padding-y: fd-space(3) !default;
+  $fd-dialog-inner-content-background: fd-color("background", 2) !default;
+
+  
+.#{$block}{
+  @include fd-reset;
+  
+  &__content{
+    border-radius: $fd-border-radius;
+    background-color: $fd-dialog-inner-content-background;
+  }
+
+  &__header {
+    position: relative;
+    border-bottom: 1px solid $fd-dialog-border-color;
+    padding: $fd-dialog-padding-y $fd-dialog-padding-x;
+  }
+  
+  &__header-error{
+    @extend .#{$block}__header;
+    border-bottom: 1px solid $fd-color--error;
+    h1::before{
+        color:$fd-color--error;
+    }
+  }
+
+  &__header-warning{
+    @extend .#{$block}__header;
+    border-bottom: 1px solid $fd-color--warning;
+    h1::before{
+        color:$fd-color--warning;
+    }
+  }
+
+  &__header-success{
+    @extend .#{$block}__header;
+    border-bottom: 1px solid $fd-color--success;
+    h1::before{
+        color:$fd-color--success;
+    }
+  }
+  
+  
+  &__title{
+    @include fd-type("1");
+    margin-bottom: 0;
+    &::before {
+        margin-right: fd-space(2);
+        vertical-align: bottom;
+      }
+      @at-root {
+        [dir="rtl"] &,
+        &[dir="rtl"] {
+          &::before {
+            margin-right: 0;
+            margin-left: fd-space(2);
+          }
+        }
+      }
+  }
+
+  &__body{
+    padding: $fd-dialog-padding-y $fd-dialog-padding-x;
+    max-height: calc(100vh - #{$fd-dialog-header-height} - #{$fd-dialog-footer-height});
+    overflow-y : auto;
+    > p {
+      padding: 0;
+    }
+  }
+  &__footer {
+    max-height: $fd-dialog-footer-height;
+    padding:  fd-space(1) $fd-dialog-padding-x; 
+    text-align: right;
+    border-top: 1px solid $fd-dialog-border-color;
+  }
+  
+}
+
+.#{$block}--small{
+    @extend .#{$block};
+    max-width: $fd-dialog-width--small;
+}
+
+.#{$block}--medium{
+    @extend .#{$block};
+    max-width: $fd-dialog-width--medium;
+}
+
+.#{$block}--phone{
+    @extend .#{$block};
+    max-width: $fd-dialog-width--phone;
+}
```

Comment:
None of these modifiers should be necessary based on the designs we have. These can be accomplished with media queries or by using max-widths.